marlowe-playground-client/src/Simulation/BottomPanel.purs

```diff
@@ -283,8 +288,13 @@ panelContents state StaticAnalysisView =
     [ classes [ ClassName "panel-sub-header", aHorizontal, Classes.panelContents ]
     ]
     [ analysisResultPane state
-    , button [ onClick $ const $ Just $ AnalyseContract, enabled (state ^. _analysisState <<< to isLoading <<< to not) ] [ text "Analyse" ]
+    , button [ onClick $ const $ Just $ AnalyseContract, enabled isEnabled, classes (if isEnabled then [] else [ ClassName "disabled" ]) ]
+        [ text (if isLoading' then "Analysing..." else "Analyse") ]
     ]
+  where
+  isLoading' = state ^. _analysisState <<< to isLoading
+
+  isEnabled = not isLoading' && isContractValid state
```

Comment:
Would it be clearer if we had `isX` being the function and `X` being the answer. So `isLoading` and `loading/enabled`? 🤔